server/client.go

```diff
@@ -146,6 +159,36 @@ func (c *client) initClient() {
 	}
 }
 
+// RegisterUser allows auth to call back into a new client
+// with the authenticated user. This is used to map any permissions
+// into the client.
+func (c *client) RegisterUser(user *User) {
+	if user.Permissions == nil {
+		return
+	}
+
+	// Process Permissions and map into client connection structures.
+	c.mu.Lock()
+	defer c.mu.Unlock()
+
+	c.perms = &permissions{}
+	c.perms.sub = NewSublist()
```

Comment:
No big deal, but should we create those only if there are permissions for sub and/or pub? I see that we later always check if `perms.sub` or `perms.pub` is not `nil` even when `perms` is not.